test/index.js

```diff
@@ -0,0 +1,17 @@
+var Lab    = require('lab')
+var assert = require('assert')
+var server = require('../lib/server')(8000)
+
+var lab = exports.lab = Lab.script()
+
+lab.test('/version endpoint returns the project version', function(done) {
+  server.inject('/version', function(response) {
+    var result = response.result
+
+    assert.ok(result instanceof Object)
+    assert.equal(response.statusCode, 200)
+    assert.equal(result.version, '0.0.1');
```

Comment:
Maybe a better test would just be to verify that it matches d+\.d+\.d+ ?
That way you never need to update it, but it will still prove the behavior.

On Tuesday, January 5, 2016, Nathan Hunzaker <notifications@github.com>
wrote:

> In test/index.js
> <https://github.com/vigetlabs/university/pull/5#discussion_r48925729>:
>
> > @@ -0,0 +1,17 @@
> > +var Lab    = require('lab')
> > +var assert = require('assert')
> > +var server = require('../lib/server')(8000)
> > +
> > +var lab = exports.lab = Lab.script()
> > +
> > +lab.test('/version endpoint returns the project version', function(done) {
> > +  server.inject('/version', function(response) {
> > +    var result = response.result
> > +
> > +    assert.ok(result instanceof Object)
> > +    assert.equal(response.statusCode, 200)
> > +    assert.equal(result.version, '0.0.1');
>
> To a degree, yes. I have difficulty providing an authoritive answer. Most
> of the behavior here relates to routing, so the duplication is okay by me.
>
> In my submission, I export the expected value from a server.js module, but
> you could also bring that up for debate as well.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vigetlabs/university/pull/5/files#r48925729>.
>


-- 
*Doug Avery*
Front-End Development Director

703.891.0670 x138
720.666.3852 m
viget.com
<http://www.google.com/url?q=http%3A%2F%2Fviget.com&sa=D&sntz=1&usg=AFrqEzcM3JYXwowZdGV1yX2CD2NWL_AHOA>

This email may include confidential information. Please do not share unless
we agree otherwise.